misc/format.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python3
+
+import os, re
+import subprocess as sp
+from pathlib import Path
+from colorama import Fore, Back, Style
+from yapf.yapflib.yapf_api import FormatFile
+from git import Repo
+
+repo_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+
+
+def has_suffix(f, suffixes):
+    for suf in suffixes:
+        if f.endswith('.' + suf):
+            return True
+    return False
+
+
+def format_plain_text(fn):
+    formatted = ''
+    with open(fn, 'r') as f:
+        for l in f:
+            l = l.rstrip()
+            if l.find('\t') != -1:
+                print(f'Warning: found tab in {fn}. Skipping...')
+                return
+            formatted += l + '\n'
+    while len(formatted) and formatted[-1] == '\n':
+        formatted = formatted[:-1]
+    formatted += '\n'
+    with open(fn, 'w') as f:
+        f.write(formatted)
+
+
+def _find_clang_format_bin():
+    candidates = ['clang-format-6.0', 'clang-format']
```

Comment:
This is probably not an issue of this PR - should we stick to `clang-format-6.0` only? Different versions of `clang-format` can give you different results.

Reply:
The action server now seems to use clang-format-6.0.
Can we only use cf6? If user can't deal with cf6 installation, they should use our format server.
This req discussion and want to talk in a sep pr anyway.